### PR TITLE
/postcode endpoint now exposes country info

### PR DESCRIPTION
**Pull request by aldavidson**
Also changed the postcode_gss_code_importer to use the offline-table method,
for a big speed-up. Refactored the reuseable parts out of the addressbase_import.sh script
to achieve this in a DRY-way. 

Not 100% happy with the separation yet - it could be cleaner, and
at some point we may want to move some of the bash functions back in to Python - but this is
good enough at this stage.

### CODE REVIEW

File: postcodeinfo/apps/postcode_api/importers/countries_importer.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+                    country = self.find_or_create_country(row['CTRY14CD'])
+                    country.name = row['CTRY14NM']
+                    country.save()
+
+    def find_or_create_country(self, gss_code):
+        try:
+            c = Country.objects.get(gss_code=gss_code)
+        except Country.DoesNotExist:
+            c = Country(gss_code=gss_code)
+        return c
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/importers/countries_importer.py]: I think you can replace this with `get_or_create` - https://docs.djangoproject.com/en/1.8/ref/models/querysets/#get-or-create

File: postcodeinfo/apps/postcode_api/importers/psql_import_adapter.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        'executing {cmd} with env {env}'.format(cmd=str(exe), env=env))
+    p = subprocess.Popen(
+        exe, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, env=env)
+    p.wait()
+
+
+class PSQLImportAdapter(object):
+
+    def import_csv(self, script_name, filepaths):
+        logging.debug("importing file(s) {filepaths}".format(filepaths=filepaths))
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/importers/psql_import_adapter.py]: line too long

File: postcodeinfo/apps/postcode_api/importers/psql_import_adapter.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        script = os.path.join(
+            settings.BASE_DIR, 'scripts/', script_name)
+        # need to explicitly pass through the DB_NAME as an env var here,
+        # because when running tests, Django automatically changes the name in settings to
+        # 'test_xyz', but *doesn't* alter the DB_NAME env var - so we have to override it
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/importers/psql_import_adapter.py]: these two lines too long

File: postcodeinfo/apps/postcode_api/importers/psql_import_adapter.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    p = subprocess.Popen(
+        exe, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, env=env)
+    p.wait()
+
+
+class PSQLImportAdapter(object):
+
+    def import_csv(self, script_name, filepaths):
+        logging.debug("importing file(s) {filepaths}".format(filepaths=filepaths))
+        script = os.path.join(
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/importers/psql_import_adapter.py]: this line doesn't need to be split

File: postcodeinfo/apps/postcode_api/importers/psql_import_adapter.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        # need to explicitly pass through the DB_NAME as an env var here,
+        # because when running tests, Django automatically changes the name in settings to
+        # 'test_xyz', but *doesn't* alter the DB_NAME env var - so we have to override it
+        # otherwise tests that try to import stuff in setup then read it back,
+        # will fail
+        env = self._db_env()
+        if not isinstance(filepaths, list):
+            filepaths = [filepaths]
+
+        runProcess(
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/importers/psql_import_adapter.py]: this line doesn't need to be split

File: postcodeinfo/apps/postcode_api/importers/psql_import_adapter.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            [script] + filepaths, env=env)
+
+    def _db_env(self):
+        return {'DB_NAME': settings.DATABASES['default']['NAME'],
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/importers/psql_import_adapter.py]: arbitrary indentation - please split the line after `{` instead

File: postcodeinfo/apps/postcode_api/management/commands/import_countries.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+from postcode_api.importers.countries_importer \
+    import CountriesImporter
+
+
+class Command(BaseCommand):
+    args = '<csv_file csv_file...>'
+
+    def handle(self, *args, **options):
+        if len(args) == 0:
```
Comment (andyhd) [inline postcodeinfo/apps/postcode_api/management/commands/import_countries.py]: it's more pythonic to say `if args:` here

**Comment by aldavidson** [comment]
all issues fixed - ready for re-review

**Comment by andyhd** [comment]
:+1:

**Pull request merged**